hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4272,6 +4275,30 @@ func reconcileVirtLauncherNetworkPolicy(policy *networkingv1.NetworkPolicy, hclu
 			},
 		},
 	}
+
+	if hcluster.Spec.Services[0].Type == hyperv1.NodePort {
+		nodeAddress, err := core.GetAPIServerAddressByNode(ctx, log.Log)
```

Comment:
I don't think this is guaranteed to get the same node address every time this function is called.  Because of this, this function is typically only used by the client side cli.


I think you can look at the hostedcluster object and find the nodeport address used for the api server. It will be in the `hc.Spec.Services` part of the api